### PR TITLE
Migrating eventgrid exntension to DI model

### PR DESCRIPTION
**Pull request by soninaren**

### CODE REVIEW

File: src/EventGridExtension/EventGridExtensionConfigProvider.cs [hunk 1]
```diff
@@ -11,31 +11,40 @@
...
         // for end to end testing
-        internal EventGridExtensionConfig(Func<EventGridAttribute, IAsyncCollector<EventGridEvent>> converter)
+        internal EventGridExtensionConfigProvider(Func<EventGridAttribute, IAsyncCollector<EventGridEvent>> converter, ILoggerFactory loggerFactory)
         {
             _converter = converter;
+            _loggerFactory = loggerFactory;
         }
 
         // default constructor
-        public EventGridExtensionConfig()
```
Comment (fabiocav) [inline src/EventGridExtension/EventGridExtensionConfigProvider.cs]: Decided against cleaning these constructors up to take an `IEventGridClient`? That would make this process cleaner and enhance the tests. If there's a good reason not to do that now, let's at least have an issue open to track.

File: src/EventGridExtension/TriggerBinding/EventGridListener.cs [hunk 2]
```diff
@@ -7,14 +7,14 @@
 
 namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
 {
-    public class EventGridListener : Host.Listeners.IListener
+    internal class EventGridListener : Host.Listeners.IListener
     {
         public ITriggeredFunctionExecutor Executor { private set; get; }
 
-        private EventGridExtensionConfig _listenersStore;
+        private EventGridExtensionConfigProvider _listenersStore;
```
Comment (mathewc) [inline src/EventGridExtension/TriggerBinding/EventGridListener.cs]: nit: renam field to _configProvider

File: src/EventGridExtension/TriggerBinding/EventGridTriggerAttributeBindingProvider.cs [hunk 3]
```diff
@@ -45,9 +45,9 @@ internal class EventGridTriggerBinding : ITriggerBinding
         {
             private readonly ParameterInfo _parameter;
             private readonly Dictionary<string, Type> _bindingContract;
-            private readonly EventGridExtensionConfig _listenersStore;
+            private readonly EventGridExtensionConfigProvider _listenersStore;
```
Comment (mathewc) [inline src/EventGridExtension/TriggerBinding/EventGridTriggerAttributeBindingProvider.cs]: same renames in here

**Comment by fabiocav** [comment]
@soninaren are we still planning on have this done for the next drop?

File: src/EventGridExtension/EventGridExtensionConfigProvider.cs (hunk 1 above)
Comment (soninaren) [reply src/EventGridExtension/EventGridExtensionConfigProvider.cs]: I am going to follow up with another PR for that change. Have updated the rest as per the PR comments.

**Pull request merged**